compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -760,6 +763,20 @@ private BLangAssignment createAssignmentToClosureMap(BLangSimpleVariableDef varD
         return ASTBuilderUtil.createAssignmentStmt(varDefNode.pos, accessExpr, varDefNode.var.expr);
     }
 
+    private BVarSymbol findClosureMapSymbol(int absoluteLevel) {
+        SymbolEnv symbolEnv = env;
+        NodeKind nodeKind = symbolEnv.node.getKind();
+        BVarSymbol mapSymbol = null;
+        while (symbolEnv != null && nodeKind != NodeKind.PACKAGE) {
```

Comment:
In this context, the check for `symbolEnv != null` is unnecessary. In line 768, we obtain the node from `symbolEnv`. Therefore, if `symbolEnv` is null, we are unable to perform that as well.